streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -685,13 +685,13 @@ public boolean test(final K1 key, final V1 value) {
                                                                                     windows.maintainMs());
 
             KStreamKStreamJoin<K1, R, V1, V2> joinThis = new KStreamKStreamJoin<>(otherWindow.name(),
-                                                                                  windows.before,
                                                                                   windows.after,
+                                                                                  windows.before,
```

Comment:
No.  From the `JoinWindows` documentation:
```sql
SELECT * FROM stream1, stream2
WHERE
  stream1.key = stream2.key
  AND
  stream2.ts - before <= stream1.ts <= stream2.ts + after
```
When `stream1.ts` is 1,000 and `before` is 100, that is satisfied when `stream2.ts` is 1,000-1,100:

```
1000 - 100 <= 1000 <= 1000 + 0
900 <= 1000 <= 1000

1100 - 100 <= 1000 <= 1100 + 0
1000 <= 1000 <= 1100
```

When `stream2.ts` is 1,000 and `before` is 100, then it is satisfied when `stream1.ts` is 900-1,000:

```
1000 - 100 <= stream1.ts <= 1000 + 0
900 <= stream1.ts <= 1000
```

I.e. `stream1.join(stream2, JoinWindows.of(0).before(100))` means _join records from stream1 that are concurrent or up to 100 milliseconds ahead (before) records from stream2_.

As a side node, the API should probably allow the specification of whether concurrent events should should be included in the `before` and `after` specification.  I.e. whether the interval to match should be open or closed on each end.